### PR TITLE
Add DebProfile to Profilemanager

### PR DESCRIPTION
**Pull request by mdellweg**

### CODE REVIEW

**Comment by mdellweg** [comment]
Still not even confirmed to work. But i'd be happy to get some feedback on this.

**Comment by mdellweg** [comment]
@kahowell As i am trying to reuse the repofile.py here, a little bit of restructuring is needed. I would like your :eyes: on this, to know this is going in the right direction.
(atm enabled_repos profile is working for AptRepoFile only, but the others should be repaired by tomorrow.)

**Comment by mdellweg** [comment]
I think, this is ready for review now.

**Comment by cnsnyder** [comment]
@mdellweg Overall looks great! I am curious about one thing. Why move repofile to rhsm?

**Comment by cnsnyder** [comment]
@candlepin-bot ok to test

**Comment by mdellweg** [comment]
> @mdellweg Overall looks great! I am curious about one thing. Why move repofile to rhsm?

I think the dependency chain is that subman uses rhsm, and since the new code, being part of rhsm and using repofile.py, that code must move up on that chain.

**Comment by mdellweg** [comment]
@cnsnyder can you tell me a bit more about the test failures, as i cannot access the results.

**Comment by cnsnyder** [comment]
@candlepin-bot please retest this

**Comment by mdellweg** [comment]
@cnsnyder rebased due to conflicts.

**Comment by mdellweg** [comment]
@cnsnyder ^^

**Comment by jturel** [comment]
@cnsnyder any review feedback for Matthias?

**Comment by cnsnyder** [comment]
@candlepin-bot Please retest this

**Comment by cnsnyder** [comment]
@candlepin-bot please retest this

File: src/subscription_manager/cache.py [hunk 1]
```diff
@@ -489,26 +476,19 @@ def _sync_with_server(self, uep, consumer_uuid, *args, **kwargs):
...
+                }
+                for key, value in combined_profile.items()
             ]
             uep.updateCombinedProfile(
                 consumer_uuid,
-                _combined_profile
+                _combined_profile,
             )
-        else:
+        elif "rpm" in combined_profile:
```
Comment (jirihnidek) [inline src/subscription_manager/cache.py]: This change of logic has potential to break functionality. Candlepin server cannot have capability "combined_reporting", but it still can be able to have capability of receiving package profile.

**Review by jirihnidek — CHANGES_REQUESTED** [review]
Please solve conflicts with master branch.

**Comment by mdellweg** [comment]
I have big trouble keeping this PR up in shape. I can not see any results of the tests, and in the meantime something in the Vagrant part must have changed to the point where i can not even provision the centos box.
It would be nice, if someone with access to the ci server logs could take this up, as i do not have the resources to more than the occasionally rebase.

**Comment by jirihnidek** [comment]
Hi @mdellweg I will take this up. Thanks for this PR.

**Comment by mdellweg** [comment]
Thank you so much! It would have been a shame to see that work going to waste.

**Comment by jturel** [comment]
Needs rebase :)

**Comment by m-bucher** [comment]
@mdellweg @jirihnidek I did a rebase to the best of my knowledge (which arguably is not much, I am afraid).

Also added a fix to only upload enabled repositories for Debian hosts

**Comment by wottop** [comment]
This needs a rebase to be considered. If no action is taken on this PR, it will be closed on August 27, 2021.

**Comment by wottop** [comment]
@m-bucher can you please rebase to main and try the nose and stylish tests locally?

**Comment by wottop** [comment]
@m-bucher @mdellweg
Can you please follow up with a rebase and then confirm the tests with local runs?
Thanks

**Comment by wottop** [comment]
@m-bucher @mdellweg
Can you please follow up with a rebase and then confirm the tests with local runs?
Thanks

**Comment by mdellweg** [comment]
@wottop Sorry, I won't have the capacity to work on this in the forseeable future.

**Comment by sbernhard** [comment]
Rebased to current main. We want to proceed with this change as it is necessary to have a full-featured debian/ubuntu subscription-manager. 

Please have a look.

**Review by ptoscano — CHANGES_REQUESTED** [review]
Sorry for the (again) late answer to this PR. I'm afraid I will need to ask for one more round of rebase on top of the current `main` branch.

In addition to that, please start cleaning this work a little bit, so it has less changes and it is easier to review:
- please avoid extra changes to the debug lines
- many `open()` calls got `encoding='utf-8'`, which I don't understand what it should do in the context of the proposed changes
- there was an recent-ish comment by @jirihnidek that I don't see addressed or explained
- please squash the commits with fixups together to the commits that they fix, otherwise there are broken commits in the history (which makes bisection and regression checking in general harder)
- please also improve the commit messages & the description of this PR: it is hard to understand why certain changes were done, and in general what is the final goal; yes, I can imagine it seems "logic" as it is already, however it will not be that much in 1 years/5 years/etc, making it harder to other developers to figure out *why* something was done

In any case, thanks again for the patience on this, and looking forward to a newer version!

**Comment by bastian-src** [comment]
The debug line and enconding changes regarding `open` were introduced due to pylint warnings on both of these. I've dropped the last commit which introduced those changes.
Thanks for the feedback on this, your other comments are gonna be worked on soon!

**Comment by cnsnyder** [comment]
@bastian-src Hi! I hope this note finds you well! Is there anything we can do to help out on these changes? If there is still more to be done and it will take some time could we close this PR and open a new one once the changes have been complete?

**Comment by cnsnyder** [comment]
@bastian-src Thank you for all your work on this PR! We'd really like to include this in subscription-manager at some point in the future! Since it's been about a month since there have been updates we will close this PR for the time being. When there are again updates to this branch based on the above feedback, please feel free to update this branch and repopen this PR (or if it makes more sense, feel free to open a new PR).

**Pull request closed without merging**